### PR TITLE
osdocs-690 entitlement updates

### PR DESCRIPTION
**Pull request by kalexand-rh**
https://jira.coreos.com/browse/OSDOCS-690

@wking, @reestr, PTAL?

This PR also fixes https://bugzilla.redhat.com/show_bug.cgi?id=1756513

### CODE REVIEW

File: installing/installing_restricted_networks/installing-restricted-networks-bare-metal.adoc [hunk 1]
```diff
@@ -92,6 +92,5 @@ include::modules/installation-complete-user-infra.adoc[leveloffset=+1]
 
 .Next steps
 
+* xref:../../telemetry/opting-out-of-telemetry.adoc#opting-out-of-telemetry[Opt out of telemetry].
```
Comment (wking) [inline installing/installing_restricted_networks/installing-restricted-networks-bare-metal.adoc]: This sounds like "and then you'll want to...".  Is there some phrasing that emphasizes the optional-ness?  Maybe "Optional next steps" for the section title?

File: modules/cluster-entitlements.adoc [hunk 2]
```diff
@@ -17,33 +17,22 @@
...
-management, including legally entitling your purchase from Red Hat, you must use
-the Telemetry service and access the
-link:https://cloud.redhat.com/openshift/install[OpenShift Infrastructure Providers] page.
+In {product-title} {product-version}, Telemetry is the component that provides metrics about cluster health and the success of updates. If your cluster is connected to the internet, Telemetry runs automatically, and your cluster is automatically entitled after installation.
 
-Because there is no disconnected subscription management, you cannot both opt
-out of sending data back to Red Hat and entitle your purchase. Support for
-disconnected subscription management might be added in future releases of
-{product-title}
+To perform subscription management, including legally entitling your purchase from Red Hat, you must access the link:https://cloud.redhat.com/openshift/install[OpenShift Infrastructure Providers] page. If you cannot send data back to Red Hat, you can perform link:https://cloud.redhat.com/openshift/register[Cluster registration] on this site.
```
Comment (wking) [inline modules/cluster-entitlements.adoc]: Nit: "If you cannot send data back to Red Hat" -> "If your cluster cannot..."?  And Maybe drop the "you must access" sentence, or elaborate on any required actions in the cluster-Telemetery-client-is-calling-in case?

File: modules/cluster-entitlements.adoc [hunk 3]
```diff
@@ -17,33 +17,22 @@
...
+
+You must have internet access to:
+
+* Access the link:https://cloud.redhat.com/openshift/install[OpenShift Infrastructure Providers] page to download the installation program
+* Access link:http://quay.io[quay.io] to obtain the packages that are required to install your cluster
+* Obtain the packages that are required to perform cluster updates
+* Access link:http://cloud.redhat.com[Red Hat's software as a service page] to perform subscription management
```
Comment (wking) [inline modules/cluster-entitlements.adoc]: This is the only access needed for entitlements (not the earlier list entries), and for this entry "you" is either the in-cluster Telemetry client (for automatic registration) or some admin/tooling hitting the registration endpoint.  It might be worth spelling that out.

File: modules/cluster-entitlements.adoc [hunk 4]
```diff
@@ -17,33 +17,22 @@
...
 
 ifdef::openshift-origin,openshift-enterprise[]
 
 [IMPORTANT]
 ====
-Your machines must have direct internet access to install the cluster.
+If you do not have direct internet access, you can perform a restricted network installation on infrastructure that you provision. During that process, you download the content that is required and use it to populate a mirror registry with the packages that you need to install a cluster and generate the installation program. With some installation types, the environment that you install your cluster i will not require internet access. Before you update the cluster, you update the content of the mirror registry.
```
Comment (wking) [inline modules/cluster-entitlements.adoc]: Seems orthogonal to entitlements?

File: modules/installation-overview.adoc [hunk 5]
```diff
@@ -77,15 +65,24 @@ of new certificates or keys, after installation.
 == Available platforms
 
 In {product-title} {product-version}, you can install a cluster that uses
-installer-provisioned infrastructure on only Amazon Web Services (AWS).
+installer-provisioned infrastructure on the following platforms:
```
Comment (wking) [inline modules/installation-overview.adoc]: OpenStack belongs in here somewhere, right?  Or is tgat still preview in 4.3?

File: modules/rhel-compute-overview.adoc [hunk 6]
```diff
@@ -6,30 +6,20 @@
...
 ====
-Swap memory is disabled on all RHEL machines that you add to your {product-title}
-cluster. You cannot enable swap memory on these machines.
+Swap memory is disabled on all RHEL machines that you add to your {product-title} cluster. You cannot enable swap memory on these machines.
 ====
 
-You must add RHEL compute machines to the cluster after you initialize the
-control plane.
\ No newline at end of file
+You must add RHEL compute machines to the cluster after you initialize the control plane.
```
Comment (wking) [inline modules/rhel-compute-overview.adoc]: Unrelated to the thrust of this PR, but maybe turn this around to make RHEL compute more clearly optional:

> You cannot add RHEL compute machines before initializing the control plane.

Or:

> You must add any RHEL compute machines to the cluster after you initialize the control plane.

File: modules/rhel-compute-overview.adoc [hunk 7]
```diff
@@ -6,30 +6,20 @@
...
+
+RHEL compute machines are not included when your cluster entitlement is calculated.
```
Comment (wking) [inline modules/rhel-compute-overview.adoc]: Really?  How is support for these nodes funded?

Comment (kalexand-rh) [reply modules/rhel-compute-overview.adoc]: That's what they said in the what's new presentation.

File: modules/cluster-entitlements.adoc (hunk 4 above)
Comment (kalexand-rh) [reply modules/cluster-entitlements.adoc]: The topics about internet access and entitlements.

File: modules/cluster-entitlements.adoc [hunk 8]
```diff
@@ -17,33 +17,20 @@
 [id="cluster-entitlements_{context}"]
 = Internet and Telemetry access for {product-title}
 
-In {product-title} {product-version}, Telemetry is the component that provides
-metrics about cluster health and the success of updates. To perform subscription
-management, including legally entitling your purchase from Red Hat, you must use
-the Telemetry service and access the
-link:https://cloud.redhat.com/openshift/install[OpenShift Infrastructure Providers] page.
+In {product-title} {product-version}, you require access to the internet to install and entitle your cluster. The Telemetry service, which runs by default to provide metrics about cluster health and the success of updates, also requires internet access. If your cluster is connected to the internet, Telemetry runs automatically, and your cluster is automatically entitled after installation.
```
Comment (reestr) [inline modules/cluster-entitlements.adoc]: There is no automatic entitlement on OpenShift Cluster Manager, the cluster is simply registered and then the customer has to assign the correct entitlements.  So we should probably replace the end of this with..

> and your cluster is registered to OpenShift Cluster Manager, entitlements would then need to be allocated after installation.

File: modules/cluster-entitlements.adoc [hunk 9]
```diff
@@ -17,33 +17,20 @@
...
-Because there is no disconnected subscription management, you cannot both opt
-out of sending data back to Red Hat and entitle your purchase. Support for
-disconnected subscription management might be added in future releases of
-{product-title}
+You must have internet access to:
+
+* Access the link:https://cloud.redhat.com/openshift[OpenShift Infrastructure Providers] page to download the installation program.
+* Access link:http://quay.io[quay.io] to obtain the packages that are required to install your cluster.
+* Obtain the packages that are required to perform cluster updates.
+* Access link:http://cloud.redhat.com[Red Hat's software as a service page] to perform subscription management and entitlement. If the cluster has internet access and you do not disable Telemetry, that service automatically entitles your cluster. If the Telemetry service cannot entitle your cluster, you must manually entitle it on the link:https://cloud.redhat.com/openshift/register[Cluster registration] page.
```
Comment (reestr) [inline modules/cluster-entitlements.adoc]: We should start to use the term OpenShift Cluster Manager:

> * Access link:https://cloud.redhat.com/openshift[OpenShift Cluster Manager] ...

File: modules/rhel-compute-overview.adoc (hunk 7 above)
Comment (kalexand-rh) [reply modules/rhel-compute-overview.adoc]: I double-checked with Tim. You have to entitle the RHEL nodes for RHEL, but they really don't count against your OCP entitlements. I've updated this statement per our discussion on Slack.

**Comment by kalexand-rh** [comment]
@jianlinliu, will you PTAL or suggest a different reviewer?

**Comment by jianlinliu** [comment]
@juzhao Is this in your test scope? If yes, pls help review.

**Comment by juzhao** [comment]
> @juzhao Is this in your test scope? If yes, pls help review.
Only 2 docs have "telemetry" keyword:
modules/cluster-entitlements.adoc 
modules/installation-overview.adoc

For the paragraphs of the above 2 docs which have "telemetry" keyword is our scope. Reviewed, they are LGTM.
For the paragraphs of the above 2 docs which don't have "telemetry" keyword and the other 3 docs is not our scope.

**Comment by juzhao** [comment]
> @juzhao Is this in your test scope? If yes, pls help review.

    Only 2 docs have "telemetry" keyword:
    modules/cluster-entitlements.adoc
    modules/installation-overview.adoc

For the paragraphs of the above 2 docs which have "telemetry" keyword is our scope. Reviewed, they are LGTM.
For the paragraphs of the above 2 docs which don't have "telemetry" keyword and the other 3 docs is not our scope.

**Comment by kalexand-rh** [comment]
@openshift/team-documentation, will you PTAL?

**Comment by kalexand-rh** [comment]
@wmengRH, will you PTAL at the change to modules/rhel-compute-overview.adoc?

File: modules/cluster-entitlements.adoc [hunk 10]
```diff
@@ -21,33 +21,18 @@
 [id="cluster-entitlements_{context}"]
 = Internet and Telemetry access for {product-title}
 
-In {product-title} {product-version}, Telemetry is the component that provides
-metrics about cluster health and the success of updates. To perform subscription
-management, including legally entitling your purchase from Red Hat, you must use
-the Telemetry service and access the
-link:https://cloud.redhat.com/openshift/install[OpenShift Infrastructure Providers] page.
+In {product-title} {product-version}, you require access to the internet to install and entitle your cluster. The Telemetry service, which runs by default to provide metrics about cluster health and the success of updates, also requires internet access. If your cluster is connected to the internet, Telemetry runs automatically, and your cluster is registered to link:https://cloud.redhat.com/openshift[Red Hat OpenShift Cluster Manager]. From there, you can allocate entitlements to your cluster.
```
Comment (bergerhoffer) [inline modules/cluster-entitlements.adoc]: registered to -> registered to the

File: modules/installation-complete-user-infra.adoc [hunk 11]
```diff
@@ -5,6 +5,13 @@
 // * installing/installing_restricted_networks/installing-restricted-networks-vsphere.adoc
 // * installing/installing_vsphere/installing-vsphere.adoc
 
+ifeval::["{context}" == "installing-restricted-networks-vsphere"]
```
Comment (bergerhoffer) [inline modules/installation-complete-user-infra.adoc]: I know AWS uses a different file, but just wanted to double check that the "register your cluster" step didn't need to be added there too.

File: modules/installation-overview.adoc [hunk 12]
```diff
@@ -77,15 +65,25 @@ of new certificates or keys, after installation.
...
+
+* Amazon Web Services (AWS)
+* Google Cloud Platform (GCP)
+* Microsoft Azure
+* OpenStack
```
Comment (bergerhoffer) [inline modules/installation-overview.adoc]: Shouldn't this be "Red Hat OpenStack Platform" since you point out Google/Microsoft, etc. for the others?

File: modules/installation-overview.adoc [hunk 13]
```diff
@@ -77,15 +65,25 @@ of new certificates or keys, after installation.
...
 
 In {product-title} {product-version}, you can install a cluster that uses
 user-provisioned infrastructure on the following platforms:
 
 * AWS
+* GCP
 * VMware vSphere
 * Bare metal
 
+With installations on user-provisioned infrastructure, each machine can have full internet access, you can place your cluster behind a proxy, or you can perform a _restricted network installation_. In a restricted network installation, you can download the images that are required to install a cluster, place them in a mirror registry, and use that data to install your cluster. While you require internet access to pull images for platform containers, with a restricted network installation on vSphere or bare metal infrastructure, your cluster machines do not require direct internet access.
```
Comment (bergerhoffer) [inline modules/installation-overview.adoc]: Is GCP UPI restricted network supported for 4.2? I don't see the docs in the GCP UPI PR. If not, then I'd be careful about generally saying that for installs on UPI that you can do a restricted network install.

File: modules/installation-overview.adoc (hunk 12 above)
Comment (kalexand-rh) [reply modules/installation-overview.adoc]: @bergerhoffer, I am not sure. The OSP docs themselves never use the full product name, and I am not sure if it's restricted to just the Red Hat version. @maxwelldb, thoughts?

Comment (maxwelldb) [reply modules/installation-overview.adoc]: AFAIK, (virtually?) none of it is limited to the Red Hat version.

Comment (bergerhoffer) [reply modules/installation-overview.adoc]: Ack. However, the release notes specify Red Hat's OpenStack, so we should probably get confirmation and sync these up.

[1] https://docs.openshift.com/container-platform/4.2/release_notes/ocp-4-2-release-notes.html

Comment (maxwelldb) [reply modules/installation-overview.adoc]: Engineering believes that our content applies to generic OpenStack as well as RHOSP, though we only support RHOSP. Verifying what PM wants, though, given that this is in the context of OCP docs. Could take beyond today.

Comment (maxwelldb) [reply modules/installation-overview.adoc]: PM wants to use Red Hat OpenStack Platform (RHOSP).

Comment (kalexand-rh) [reply modules/installation-overview.adoc]: Then this PR is good. Thank you for pointing this out @bergerhoffer!

**Comment by kalexand-rh** [comment]
@xltian, who should finish this review?

**Comment by jianlinliu** [comment]
@gpei could you help review the change to modules/rhel-compute-overview.adoc

**Comment by jianlinliu** [comment]
The change to modules/installation-overview.adoc  looks good to me.

**Comment by gpei** [comment]
Most of the changes to modules/rhel-compute-overview.adoc lgtm, except for the following sentence: 
`Although you must entitle them for their use of RHEL, RHEL compute machines are exempt from the {product-title} entitlement calculations.`
I don't know what does this mean, maybe it's not a functional feature covered by OpenShift QE.

**Comment by vikram-redhat** [comment]
Thanks @gpei . In this case, I will hold off on merging this and this can come as an async update tomorrow.

**Comment by kalexand-rh** [comment]
@gpei is right, and Tim Rees approved that line separately in Slack. I'm going to go ahead and merge. Thank you all for reviewing the change!
